libs/traits/src/liquidity_pools.rs

```diff
@@ -129,3 +142,38 @@ pub trait InboundMessageHandler {
 	/// Handle an inbound message.
 	fn handle(sender: Self::Sender, msg: Self::Message) -> DispatchResult;
 }
+
+/// The behavior of an entity that can forward outbound messages
+pub trait ForwardMessageSender {
+	/// The middleware by where this message is sent
+	type Middleware;
+
+	/// The originator of the message to be sent
+	type Origin;
+
+	/// The message that will be forwarded
+	type Message;
+
+	/// Forwards a message for origin to destination
+	fn forward(
+		middleware: Self::Middleware,
+		origin: Self::Origin,
+		message: Self::Message,
+	) -> DispatchResult;
+}
+
+/// The behavior of an entity that can forward received messages
+pub trait ForwardMessageReceiver {
+	/// The middleware by where this message is received
+	type Middleware;
+
+	/// The originator of the received message
+	type Origin;
+
+	/// Forwards a received message from origin
+	fn forward(
+		middleware: Self::Middleware,
+		origin: Self::Origin,
+		message: Vec<u8>,
+	) -> DispatchResult;
+}
```

Comment:
Why not reuse the `MessageSender` and `MessageReceiver`?

Reply:
Is it because the `Message` type here is as `Message`? Maybe then the message can be also an associated types in `MessageSender` and `MessageReceiver`, to reusing them? Just thinking out laud

Reply:
Yeah that is one reason. Another is that this was a proposal from @cdamian which I liked to distinguish between both traits and its implementers. Not sure what is preferable: Composability or explicitness.

Reply:
What you think is better! I think we could do that refactor if needed after the audit. So we can focus in the solution

Reply:
Combined: [e5dedb2](https://github.com/centrifuge/centrifuge-chain/pull/1971/commits/e5dedb2f696e470abd94fe3481ab6cb9dd316ca9)